lib/Platform/Intl/PlatformIntlApple.mm

```
@@ -192,6 +192,113 @@ LocaleMatch lookupMatcher(
+  auto value = options.find(property);
+  // 3. If value is undefined, return fallback.
+  if (value == options.end()) {
+    return fallback;
+  }
+  // 4. Assert: type is "boolean" or "string".
+  // 5. If type is "boolean", then
+  // a. Set value to ! ToBoolean(value).
+  // 6. If type is "string", then
+  // a. Set value to ? ToString(value).
+  // 7. If values is not undefined and values does not contain an element equal
+  // to value, throw a RangeError exception.
+  if (!value->second.isString() ||
+      (!values.empty() &&
+       llvh::find(values, value->second.getString()) == values.end())) {
+    return runtime->raiseRangeError(
+        vm::TwineChar16("Value ") +
+        vm::TwineChar16(value->second.getString().c_str()) +
+        vm::TwineChar16(
+            " out of range for Intl.DateTimeFormat options property ") +
+            vm::TwineChar16(property.c_str()));
+  }
+  // 8. Return value.
+  return llvh::Optional<std::u16string>(value->second.getString());
+}
+vm::CallResult<llvh::Optional<bool>> getOptionBool(
+    const Options &options,
+    const std::u16string &property,
+    llvh::Optional<bool> fallback,
+    vm::Runtime *runtime) {
+  //  1. Assert: Type(options) is Object.
+  //  2. Let value be ? Get(options, property).
+  auto value = options.find(property);
+  //  3. If value is undefined, return fallback.
+  if (value == options.end()) {
+    return fallback;
+  }
+  if (!value->second.isBool()) {
+    return runtime->raiseRangeError(
+        vm::TwineChar16("Value ") +
+        vm::TwineChar16(value->second.getBool()) +
+        vm::TwineChar16(
+            " out of range for Intl.DateTimeFormat options property ") +
+            vm::TwineChar16(property.c_str()));
+  }
+  //  8. Return value.
+  return llvh::Optional<bool>(value->second.getBool());
+}
+// https://tc39.es/ecma402/#sec-defaultnumberoption
+vm::CallResult<llvh::Optional<uint8_t>> defaultNumberOption(
+    const double value,
```

This argument should be an `llvh::Optional<Option>`. If the option does not contain a number, we should throw an exception, as we do in other places.

---

I tried using llvh::Optional<Option> value, but this meant I needed to pass in value->second, which causes a fault when empty, and the idea was to check for the need for a fallback in defaultNumberOption? So I changed it to an unordered map iterator. Although this also meant that I had to pass options to defaultNumberOptions to check if it is empty. Sorry, I think I might have misunderstood what you meant.